### ISSUE TITLE
False positive: `expl_impl_clone_on_copy` with `unsafe` fields.

### ISSUE THREAD

**Issue by TimNN** [rust-lang/rust-clippy#14558]
### Summary

`unsafe` fields (still unstable, see https://github.com/rust-lang/rust/issues/132922) cannot (currently) derive `Clone`.

(Feel free to close if this should wait until https://github.com/rust-lang/rust/issues/132922 is closer to being stabilized).

### Lint Name

expl_impl_clone_on_copy

### Reproducer

I tried this code:

```rust
#![feature(unsafe_fields)]
#![allow(incomplete_features)]
#![warn(clippy::pedantic)]

pub struct Valid {
    pub unsafe actual: (),
}

unsafe impl Copy for Valid {}

impl Clone for Valid {
    #[inline]
    fn clone(&self) -> Self {
        *self
    }
}
```

I saw this happen:

```
warning: you are implementing `Clone` explicitly on a `Copy` type
```

I expected to see this happen: No warnings, since attempting to derive `Clone` produces a compiler error.

### Version

```text
rustc 1.88.0-nightly (17ffbc81a 2025-04-04)
binary: rustc
commit-hash: 17ffbc81a30c094193836a5d7f90dff273b5df93
commit-date: 2025-04-04
host: aarch64-apple-darwin
release: 1.88.0-nightly
LLVM version: 20.1.2
```
